evidence/pool_test.go

```diff
@@ -195,6 +202,55 @@ func TestEvidencePoolNewPool(t *testing.T) {
 	assert.EqualValues(t, 0, pool.ValidatorLastHeight([]byte("non-existent-validator")))
 }
 
+func TestAddingAndPruningPOLC(t *testing.T) {
+	var (
+		valAddr      = []byte("validator_address")
+		stateDB      = initializeValidatorState(valAddr, 1)
+		evidenceDB   = dbm.NewMemDB()
+		blockStoreDB = dbm.NewMemDB()
+		state        = sm.LoadState(stateDB)
+		blockStore   = initializeBlockStore(blockStoreDB, state, valAddr)
+		height       = state.ConsensusParams.Evidence.MaxAgeNumBlocks * 2
+		evidenceTime = time.Date(2019, 1, 1, 0, 0, 0, 0, time.UTC)
+	)
+
+	pubKey, _ := types.NewMockPV().GetPubKey()
+	polc := types.NewMockPOLC(1, evidenceTime, pubKey)
+
+	pool, err := NewPool(stateDB, evidenceDB, blockStore)
+	require.NoError(t, err)
+
+	err = pool.AddPOLC(polc)
+	assert.NoError(t, err)
+
+	// should be able to retrieve polc
+	newPolc, err := pool.RetrievePOLC(1, 1)
+	assert.NoError(t, err)
+	assert.True(t, polc.Equal(newPolc))
+
+	// should not be able to retrieve
+	emptyPolc, err := pool.RetrievePOLC(2, 1)
+	assert.Error(t, err)
+	assert.Equal(t, types.ProofOfLockChange{}, emptyPolc)
+
+	lastCommit := makeCommit(height-1, valAddr)
+	block := types.MakeBlock(height, []types.Tx{}, lastCommit, []types.Evidence{})
+	// update state (partially)
+	state.LastBlockHeight = height
+	pool.state.LastBlockHeight = height
+
+	// update should prune the polc
+	pool.Update(block, state)
+
+	fmt.Println(pool.IsExpired(newPolc.Height(), newPolc.Time()))
```

Comment:
no Print in tests please. use `t.Log`

Reply:
whoops that would have been for me to check something probably - good catch